osutil/syncdir.go

```diff
@@ -100,21 +100,24 @@ func EnsureDirStateGlobs(dir string, globs []string, content map[string]*FileSta
 		changed = append(changed, baseName)
 	}
 	// Delete phase (remove files matching the glob that are not in content)
-	var matches []string
+	matches := make(map[string]bool)
```

Comment:
IIRC maps allocations on insert are costly. You could try adding a hint on the length, eg. `make(map[string]bool, 10)`, if this will be called frequently.

Reply:
Typically it will deal with 2-3 files but we don't know at this stage.